iscript/src/iscript/macvpn.py

```diff
@@ -64,20 +65,123 @@ async def _sign_app(config, sign_config, app, entitlements_url, provisionprofile
     """
     # We mock the task for downloading entitlements and provisioning profiles
     entitlements_path = await download_entitlements_file(config, sign_config, {"payload": {"entitlements-url": entitlements_url}})
+
+    # TODO: Add provisionprofile_dir to scriptworker config
+    # https://github.com/mozilla-releng/scriptworker/blob/master/src/scriptworker/constants.py#L34
     provisioning_profile_path = None
-    if config.get("provisionprofile_dir"):
-        provisioning_profile_path = os.path.join(config["provisionprofile_dir"], provisionprofile_filename)
+
+    if provisionprofile_filename:
+        provisioning_profile_path = os.path.join("/builds/scriptworker/provisionprofiles", provisionprofile_filename)
         if not os.path.exists(provisioning_profile_path):
             log.error(f"Could not find provisionprofile file: {provisioning_profile_path}")
             raise IScriptError(f"Could not find provisionprofile file: {provisioning_profile_path}")
-        else:
-            log.info(f"Using provisionprofile {provisioning_profile_path}")
+        log.info(f"Using provisionprofile {provisioning_profile_path}")
 
     await sign_all_apps(config, sign_config, entitlements_path, [app], provisioning_profile_path)
 
     log.info(f"Done signing app {app.app_name}")
 
 
+async def _sign_util(sign_config, binary_path):
+    """Sign command for VPN util binaries.
+
+    Args:
+        sign_config (dict): the task config
+        binary_path (str): the binary name
+    """
+    sign_command = [
+        "codesign",
+        "--timestamp",
+        "-fv",
+        "-s",
+        sign_config["identity"],
+        "--keychain",
+        sign_config["signing_keychain"],
+        "--options",
+        "runtime",
+        binary_path,
+    ]
+    await retry_async(
+        run_command,
+        args=(sign_command,),
+        kwargs={"cwd": os.path.dirname(binary_path), "exception": IScriptError, "output_log_on_exception": True},
+        retry_exceptions=(IScriptError,),
+    )
+
+
+async def _create_pkg_files(config, sign_config, app):
+    """Create .pkg installer from the .app file. VPN specific behavior.
```

Comment:
Could you elaborate what is different between this function and the one in `mac.py`? Is there any way we could have this one wrap that one instead of outright replacing it?

Reply:
The arguments for the underlying commands are quite different. `mac.py` has options that work for gecko, and making it "generic" doesn't mean the next app will be able to use it either.

Reply:
One way we solved for this was using [MAC_PRODUCT_CONFIG](https://github.com/mozilla-releng/scriptworker-scripts/blob/master/iscript/src/iscript/constants.py#L4-L26); we could get our commands / arguments from that config dictionary, if our workflow is otherwise similar.